openflow15/action.go

```diff
@@ -116,13 +119,18 @@ func DecodeAction(data []byte) (Action, error) {
 		}
 		v := binary.BigEndian.Uint32(data[4:8])
 		if v == NxExperimenterID {
-			a = DecodeNxAction(data)
+			a, err = DecodeNxAction(data)
+			if err != nil {
+				klog.V(4).InfoS("Failed to decode NxAction", "err", err, "data", data)
```

Comment:
If the target is to log the error, why not use `klog.ErrorS`? The same as other calls in this change.

Reply:
Another question is, I think the error is possible logged in the caller, so it is possible to introduce a replicated log eventually. Instead, we should enhace the error with a wrapper to say what error exists, for example,

```
			if err != nil {
				return fmt.Errorf("failed to decode NxAction on data %v: %v", data, err)
                        }
```

What do you think?

Reply:
Yes, I like both methods you suggest.

In some places where we want to dump data, I think klog.ErrorS is better. Otherwise, if we wrap the data in the error, we don't know whether the caller will print out the error.

What do you think?

Reply:
I am OK that logging the errors in libOpenflow project directly, if the error is important. Just a clarification that, antrea as a consumer of libOpenlow, the errors will be printed eventually whenever it happens. In antrea code, we are trying to not ignore any error.